archive_converter.py

```diff
@@ -13,8 +13,57 @@
 # limitations under the License.
 
 import email
+import os
 from setup_gmail import Message
 
+class ArchiveMessageIndex(object):
+    def __init__(self):
+        self._lookup_index = {} # Maps MessageKey to Thread
```

Comment:
It says "MessageKey" in the comment, but based on `find_thread`'s type annotation, I assume the key is `str`?

There is a "MessageKey" type, so I'm not sure which one of these to trust.
Can you update the comment (or using tying to make it clear?)